### ISSUE TITLE
[ANDROID] Unable to pause activity Error

### ISSUE THREAD

**Issue by eul721** [facebook/react-native#10080]
### Issue Description

I've been getting errors when I build for my Android device. 
At first I thought it was a library error ( [react-native-image-crop-picker](https://github.com/ivpusic/react-native-image-crop-picker) specifically) 
However I started getting this exact error occasionally when I build using `react-native run-android` It's more stable when I run it from Android Studio, but sometimes I still get it.

```
FATAL EXCEPTION: main
Process: com.myapp, PID: 14470
java.lang.RuntimeException: Unable to pause activity {com.myapp/com.myapp.MainActivity}: android.os.NetworkOnMainThreadException
  at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:3722)
  at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3688)
  at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3662)
  at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:3636)
  at android.app.ActivityThread.-wrap16(ActivityThread.java)
  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1472)
  at android.os.Handler.dispatchMessage(Handler.java:102)
  at android.os.Looper.loop(Looper.java:154)
  at android.app.ActivityThread.main(ActivityThread.java:6077)
  at java.lang.reflect.Method.invoke(Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755)
Caused by: android.os.NetworkOnMainThreadException
  at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1303)
  at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:111)
  at java.net.SocketOutputStream.write(SocketOutputStream.java:157)
  at okio.Okio$1.write(Okio.java:78)
  at okio.AsyncTimeout$1.write(AsyncTimeout.java:180)
  at okio.RealBufferedSink.emit(RealBufferedSink.java:178)
  at okhttp3.internal.ws.WebSocketWriter.writeControlFrameSynchronized(WebSocketWriter.java:152)
  at okhttp3.internal.ws.WebSocketWriter.writeClose(WebSocketWriter.java:111)
  at okhttp3.internal.ws.RealWebSocket.close(RealWebSocket.java:168)
  at com.facebook.react.devsupport.JSPackagerWebSocketClient.closeWebSocketQuietly(JSPackagerWebSocketClient.java:104)
  at com.facebook.react.devsupport.JSPackagerWebSocketClient.closeQuietly(JSPackagerWebSocketClient.java:98)
  at com.facebook.react.devsupport.DevServerHelper.closePackagerConnection(DevServerHelper.java:129)
  at com.facebook.react.devsupport.DevSupportManagerImpl.reload(DevSupportManagerImpl.java:846)
  at com.facebook.react.devsupport.DevSupportManagerImpl.setDevSupportEnabled(DevSupportManagerImpl.java:481)
  at com.facebook.react.XReactInstanceManagerImpl.onHostPause(XReactInstanceManagerImpl.java:490)
  at com.facebook.react.XReactInstanceManagerImpl.onHostPause(XReactInstanceManagerImpl.java:504)
  at com.facebook.react.ReactActivityDelegate.onPause(ReactActivityDelegate.java:110)
  at com.facebook.react.ReactActivity.onPause(ReactActivity.java:60)
  at android.app.Activity.performPause(Activity.java:6804)
  at android.app.Instrumentation.callActivityOnPause(Instrumentation.java:1322)
  at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:3711)
  at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3688) 
  at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3662) 
  at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:3636) 
  at android.app.ActivityThread.-wrap16(ActivityThread.java) 
  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1472) 
  at android.os.Handler.dispatchMessage(Handler.java:102) 
  at android.os.Looper.loop(Looper.java:154) 
  at android.app.ActivityThread.main(ActivityThread.java:6077) 
  at java.lang.reflect.Method.invoke(Native Method)
```

Strange enough, this only happens on my Nexus 6P (which updated to Android 7.0 last week). 

FYI here are the libraries I'm using:

```
"react": "15.3.1",
"react-native": "^0.34.0",
"react-native-gifted-listview": "^0.0.5",
"react-native-image-crop-picker": "^0.9.5",
"react-native-material-kit": "^0.3.2",
"react-native-navbar": "^1.5.0",
"react-native-router-flux": "^3.35.0",
"react-native-star-rating": "^1.0.6",
"react-native-vector-icons": "^2.1.0",
```
### Additional Information
- React Native version: 0.34
- Platform(s) (iOS, Android, or both?): Android
- Operating System (macOS, Linux, or Windows?): MacOS

**Comment by sarriaroman** [comment]
Happening on Linking module too since 0.34

**Comment by kendrickkesley** [comment]
Same for me.. Always error on Linking module
Sometime the error occurs on start up

**Comment by wangcheng714** [comment]
me to, after update to 0.34

**Comment by sarriaroman** [comment]
Ugly solution. Override onCreate in MainActivity.java

```java
@Override
    protected void onCreate(Bundle savedInstanceState) {
        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
        StrictMode.setThreadPolicy(policy);

        super.onCreate(savedInstanceState);
    }
```

**Comment by kendrickkesley** [comment]
@sarriaroman the linking module still crashes with that :( 
And I just discovered back button recent button will also crash the app

**Comment by kkgelu** [comment]
Same for me, I started another activity (on top of the main activity) through a JS->Native call. It's on Android 7.

**Comment by ghuh** [comment]
Ditto.  Android 7 emulator.

**Comment by eul721** [comment]
Is everybody running on Android 7 when they encountered this error?

**Comment by tonypeng** [comment]
Same, issue started happening after upgrading to RN 0.34 and updating my device to Nougat. Getting the "app_name has stopped" popup whenever leaving the app (e.g. home button, opening recent apps, etc.)

Logcat:
```
8 8901-8901/com.app E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.app, PID: 8901
    java.lang.RuntimeException: Unable to pause activity {com.app/com.app.MainActivity}: android.os.NetworkOnMainThreadException
        at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:3722)
        at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3688)
        at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3662)
        at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:3636)
        at android.app.ActivityThread.-wrap16(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1472)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6077)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755)
     Caused by: android.os.NetworkOnMainThreadException
        at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1303)
        at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:111)
        at java.net.SocketOutputStream.write(SocketOutputStream.java:157)
        at okio.Okio$1.write(Okio.java:78)
        at okio.AsyncTimeout$1.write(AsyncTimeout.java:180)
        at okio.RealBufferedSink.emit(RealBufferedSink.java:178)
        at okhttp3.internal.ws.WebSocketWriter.writeControlFrameSynchronized(WebSocketWriter.java:152)
        at okhttp3.internal.ws.WebSocketWriter.writeClose(WebSocketWriter.java:111)
        at okhttp3.internal.ws.RealWebSocket.close(RealWebSocket.java:168)
        at com.facebook.react.devsupport.JSPackagerWebSocketClient.closeWebSocketQuietly(JSPackagerWebSocketClient.java:104)
        at com.facebook.react.devsupport.JSPackagerWebSocketClient.closeQuietly(JSPackagerWebSocketClient.java:98)
        at com.facebook.react.devsupport.DevServerHelper.closePackagerConnection(DevServerHelper.java:129)
        at com.facebook.react.devsupport.DevSupportManagerImpl.reload(DevSupportManagerImpl.java:846)
        at com.facebook.react.devsupport.DevSupportManagerImpl.setDevSupportEnabled(DevSupportManagerImpl.java:481)
        at com.facebook.react.XReactInstanceManagerImpl.onHostPause(XReactInstanceManagerImpl.java:490)
        at com.facebook.react.XReactInstanceManagerImpl.onHostPause(XReactInstanceManagerImpl.java:504)
        at com.facebook.react.ReactActivityDelegate.onPause(ReactActivityDelegate.java:110)
        at com.facebook.react.ReactActivity.onPause(ReactActivity.java:60)
        at android.app.Activity.performPause(Activity.java:6804)
        at android.app.Instrumentation.callActivityOnPause(Instrumentation.java:1322)
        at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:3711)
        at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3688) 
        at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3662) 
        at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:3636) 
        at android.app.ActivityThread.-wrap16(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1472) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6077) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755) 
```

**Comment by chrisknepper** [comment]
Possibly related to [changes](https://developer.android.com/about/versions/nougat/android-7.0-changes.html) in Nougat (Android 7.0)?

**Comment by Kerumen** [comment]
Hum it seems.. In the[ Other Important Points](https://developer.android.com/about/versions/nougat/android-7.0-changes.html#other) section:

>Due to a bug in previous versions of Android, the system did not flag writing to a TCP socket on the main thread as a strict-mode violation. Android 7.0 fixes this bug. Apps that exhibit this behavior now throw an android.os.NetworkOnMainThreadException. Generally, performing network operations on the main thread is a bad idea because these operations usually have a high tail latency that causes ANRs and jank.

**Comment by AndrewJack** [comment]
From what I can tell from the stack trace it looks like this will only occur when dev mode is enabled.

It was introduced by this commit: 
https://github.com/facebook/react-native/commit/588f0b83e15df0a8ef3930c4a02f4fe4087ab147

Then reverted:
https://github.com/facebook/react-native/commit/027a2706d2e1523508404f13ebb6f9b0403d86e2

Then reapplied and fixed in 0.35.0-rc:
https://github.com/facebook/react-native/commit/16bdbee165b41d58e6ef387fe30d5d9a0abed711

It looks like @foghina has already fixed in 0.35.0-rc by [wrapping the close websocket in an AsyncTask](https://github.com/facebook/react-native/commit/16bdbee165b41d58e6ef387fe30d5d9a0abed711#diff-48d7d48e38fb100b8cbe13ca6d5e1671R134)

**Comment by AndrewJack** [comment]
@facebook-github-bot close

Fixed in 0.35.0 closing issue.

**Comment by dsibiski** [comment]
Closing as it seems like the github-bot was down before...
